### DISCUSSION TITLE
using ginan for post-processing of a user rinex files

### DISCUSSION THREAD

**Discussion by pktrigg** [GeoscienceAustralia/ginan#97] [General]
Hi, we collect rinex files on buoys offshore for the Au Government.  we would like to use ginan to post-process these runex files to compute .pos position+quality files.
At present we are using NRCan but we would prefer to use ginan if possible.

I took your ppp_example.yaml configuration and made the folowing changes...

configured to read out rinex files.
manually downloaded the ppp products from GA FTP site for the corresponding day
attempted to run pea as follows

`../bin/pea -y ../exampleConfigs/guardian_example.yaml
`

- The software runs ok
- it appears to load the products files ok
- it appears to be reading the epochs from the input rinex file ok
- unfortunately I do not get a position solution computed.  I was hoping we would get a PPP solution computed.
- the output from pea is attached.  

**Is it possible to use ginan to post-process a rinex file and compute position for each epoch a ppp solution and output to a regualr .pos file as we would with NRCAN!**

[ginan_pea_output.txt](https://github.com/user-attachments/files/16310445/ginan_pea_output.txt)

[guardian_example.zip](https://github.com/user-attachments/files/16310397/guardian_example.zip)

**Comment by smcclusky** [comment]
Hi Pktrigg,
Yes - Ginan will give you a position estimate for each epoch of data processed.
I had a quick look at your config file and things look reasonable as far as I can tell without actually seeing the Ginan processing output.
You will note at the top of your terminal output the following `Warning: outputs:ppp_sol: is not a valid yaml option`
This is because the `.pos` format output has been removed in the latest version of Ginan. As an alternative you can get the coordinates from the `.gpx` file (lat/lon/e_ht), or the cartesian coordinates from the `.trace` file that should have also been output in your run?
Also I see you are missing an input bias file `bsx_files:` that will be needed to obtain the satellite code biases needed for this run?
You can use the `auto_download_PPP.py` script provided with ginan to get these for you, or find them at the CDDIS GNSS products archive?
I would be happy to work with you to optimise a ginan workflow for your application. Feel free to email me directly at simon.mcclusky@ga.gov.au

**Reply by pktrigg** [reply]
hi, thanks for getting back to me.  I was stuck on another project yesterday.  I will try out the great feedback above and see where I land.  for sure I will email you to get things going.  
The `.pos` files are a deliverable to the AU dept defence so if I can be bold and make a request it would be to have then re-instated.  if not I will need to try create one with a python script to parse the `.trace` files. possible?
